packages/design-system/src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,189 @@
+			const origin = items[0];
+			const suffix = items.slice(-2);
+			const collapsed = items.slice(1, items.length - 2);
+
+			return (
+				<>
+					<li className={styles.entry}>
+						<StackHorizontal gap="S" align="center" wrap="nowrap">
+							{'href' in origin ? (
+								<Link href={origin.href} target={origin.target} withEllipsis>
+									{origin.label}
+								</Link>
+							) : (
+								<Link as={origin.as} target={origin.target} withEllipsis>
+									{origin.label}
+								</Link>
+							)}
+
+							<span aria-hidden className={styles.divider}>
+								<Divider orientation="vertical" />
+							</span>
+						</StackHorizontal>
+					</li>
+
+					<li className={classnames(styles.entry, styles.collapsed)}>
+						<StackHorizontal gap="S" align="center" wrap="nowrap">
+							<Dropdown
+								aria-label={t('COLLAPSED_LINKS_MENU', 'Collapsed links')}
+								items={collapsed.map(collapsedLinks => {
+									if ('href' in collapsedLinks) {
+										return {
+											href: collapsedLinks.href,
+											label: collapsedLinks.label,
+											target: collapsedLinks.target,
+											type: 'link',
+										};
+									}
+									return {
+										label: collapsedLinks.label,
+										target: collapsedLinks.target,
+										type: 'link',
+										as: collapsedLinks.as,
+									};
+								})}
+							>
+								<ButtonTertiary isDropdown size="S" onClick={() => {}}>
+									…
+								</ButtonTertiary>
+							</Dropdown>
+							<span aria-hidden className={styles.divider}>
+								<Divider orientation="vertical" />
+							</span>
+						</StackHorizontal>
+					</li>
+
+					{suffix.map((entry, index) => {
+						const isLastEntry = index === suffix.length - 1;
+						if ('href' in entry) {
+							const { label, ...entryProps } = entry;
+							return (
+								<li className={styles.entry} key={`${label}-${index}`}>
+									<StackHorizontal gap="S" align="center" wrap="nowrap">
+										<Link
+											{...entryProps}
+											aria-current={isLastEntry ? 'page' : undefined}
+											withEllipsis
+										>
+											{label}
+										</Link>
+										{!isLastEntry && (
+											<span aria-hidden className={styles.divider}>
+												<Divider orientation="vertical" />
+											</span>
+										)}
+									</StackHorizontal>
+								</li>
+							);
+						}
+						const { label, as, ...entryProps } = entry;
+						return (
+							<li className={styles.entry} key={`${label}-${index}`}>
+								<StackHorizontal gap="S" align="center" wrap="nowrap">
+									<Link
+										{...entryProps}
+										as={as}
+										aria-current={isLastEntry ? 'page' : undefined}
+										withEllipsis
+									>
+										{label}
+									</Link>
+									{!isLastEntry && (
+										<span aria-hidden className={styles.divider}>
+											<Divider orientation="vertical" />
+										</span>
+									)}
+								</StackHorizontal>
+							</li>
+						);
```

Comment:
I think you can make it simpler by removing the duplication.
The only diff I see between the if and else is the `as` prop. But you don't do anything with it, just destructure it to pass it as is. This can go ini the `if` case as you already spread the props.

Reply:
But if it is to prevent having `href` AND `as`, you should remove the as in the if. But I don't think we should forbid that if the as component needs an `href` prop

Reply:
The if else ensure typesafety.

Reply:
ok 👍  so you can avoid passing the `as` in the if